packages/http-server-javascript/src/scripts/scaffold/bin.mts

```diff
@@ -0,0 +1,698 @@
+    {
+      type: "node",
+      request: "launch",
+      name: "Launch Program",
+      program: "${workspaceFolder}/dist/src/index.js",
+      preLaunchTask: "npm: build",
+      internalConsoleOptions: "neverOpen",
+    },
+  ],
+};
+
+const VSCODE_TASKS_JSON = {
+  version: "2.0.0",
+  tasks: [
+    {
+      type: "npm",
+      script: "build",
+      group: "build",
+      problemMatcher: [],
+      label: "npm: build",
+      presentation: {
+        reveal: "silent",
+      },
+    },
+  ],
+};
+
+interface ScaffoldingOptions {
+  /**
+   * If true, the project will be generated in the current directory instead of the output directory.
+   */
+  "no-standalone": boolean;
+  /**
+   * If true, writes will be forced even if the file or setting already exists. Use with caution.
+   */
+  force: boolean;
+}
+
+const DEFAULT_SCAFFOLDING_OPTIONS: ScaffoldingOptions = {
+  "no-standalone": false,
+  force: false,
+};
+
+function parseScaffoldArguments(args: string[]): ScaffoldingOptions {
+  let cursor = 2;
+  const options: Partial<ScaffoldingOptions> = {};
+
+  while (cursor < args.length) {
+    const arg = args[cursor];
+
+    if (arg === "--no-standalone") {
+      options["no-standalone"] = true;
+    } else if (arg === "--force") {
+      options.force = true;
+    } else {
+      console.error(`[hsj] Unrecognized scaffolding argument: '${arg}'`);
+      process.exit(1);
+    }
+
+    cursor++;
+  }
+
+  return { ...DEFAULT_SCAFFOLDING_OPTIONS, ...options };
+}
+
+async function confirmYesNo(message: string): Promise<void> {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  try {
+    const response = await rl.question(`${message} [y/N] `);
+
+    if (response.trim().toLowerCase() !== "y") {
+      console.error("[hsj] Operation cancelled.");
+      process.exit(0);
+    }
+  } finally {
+    rl.close();
+  }
+}
+
+export async function scaffold(options: ScaffoldingOptions) {
+  if (options.force) {
+    await confirmYesNo(
+      "[hsj] The `--force` flag is set and will overwrite existing files and settings that may have been modified. Continue?",
+    );
+  }
+
+  const cwd = process.cwd();
+
+  const projectYamlPath = path.resolve(cwd, COMMON_PATHS.projectYaml);
+  const mainTspPath = path.resolve(cwd, COMMON_PATHS.mainTsp);
+
+  console.info("[hsj] Scaffolding TypeScript project...");
+  console.info(
+    `[hsj] Using project file '${path.relative(cwd, projectYamlPath)}' and main file '${path.relative(cwd, mainTspPath)}'`,
+  );
+
+  let config: any;
+
+  try {
+    const configText = await fs.readFile(projectYamlPath);
+
+    config = YAML.parse(configText.toString("utf-8"));
+  } catch {
+    console.error(
+      "[hsj] Failed to read project configuration file. Is the project initialized using `tsp init`?",
+    );
+    process.exit(1);
+  }
+
+  // TODO: all of this path handling is awful. We need a good API from the compiler to interpolate the paths for us and
```

Comment:
It also feels like we should be using the compiler's fs, so that all of this can be easily mocked and tested